src/main.c

```diff
@@ -1755,11 +1763,7 @@ static void set_quic_sockopts(int fd, int family, unsigned sndbuf, unsigned rcvb
         break;
     }
 
-    /* set sndbuf & rcvbuf */
-    if (sndbuf != 0 && setsockopt(fd, SOL_SOCKET, SO_SNDBUF, &sndbuf, sizeof(sndbuf)) != 0)
-        h2o_fatal("failed to set SO_SNDBUF:%s", strerror(errno));
-    if (rcvbuf != 0 && setsockopt(fd, SOL_SOCKET, SO_RCVBUF, &rcvbuf, sizeof(rcvbuf)) != 0)
-        h2o_fatal("failed to set SO_RCVBUF:%s", strerror(errno));
+    set_socket_buffers(fd, sndbuf, rcvbuf);
```

Comment:
Yeah, these lines of code are executed during startup, therefore we are refusing to boot if they fail.

That's the intent of the code, we cannot change the behavior.

Reply:
Right! So extracting `set_socket_buffers()` does not make sense. 

The original patch intends to check the call of `setsockopt()`, not sharing the code with elsewhere, so I've made another one: https://github.com/h2o/h2o/pull/3116